modules/optimization/src/executioners/AdjointSolve.C

```diff
@@ -52,6 +53,14 @@ AdjointSolve::AdjointSolve(Executioner & ex)
     paramError("forward_system", "Forward system does not appear to be a 'NonlinearSystem'.");
   if (!dynamic_cast<NonlinearSystem *>(&_nl_adjoint))
     paramError("adjoint_system", "Adjoint system does not appear to be a 'NonlinearSystem'.");
+  // Adjoint system should never perform it's own automatic scaling. Scaling is
+  // taken from the forward system
+  _nl_adjoint.automaticScaling(false);
+
+  // We need to force the forward system to have a scaling vector. This is
+  // incase a user provides scaling for an individual variables but doesn't have any
+  // AD objects.
```

Comment:
why do AD objects matter for scaling?

Reply:
There are multiple ways in MOOSE to having scaling for a system. Almost all of the methods use the scaling vector except if a user supplies scaling to individual variables and has no AD objects at all. If they have a single AD object then it activates the scaling vector. Automatic scaling option in `Executioner` also activates the scaling vector. Here I am forcing the scaling vector to be turned so that the scaling worked for the singular case I mentioned above.